### ISSUE TITLE
IsInEnum not validating correctly

### ISSUE THREAD

**Issue by Arthur-Neto** [JeremySkinner/FluentValidation#1212]
### System Details

- FluentValidation version: FluentValidation.AspNetCore 8.4.0
- Web Framework version ASP.NET Core 2.2

### Issue Description
Hi, I'm trying to validate a command: 
```
public class FeatureExampleAddCommand
{
    public FeatureExampleEnum FeatureExampleType { get; set; }
}

public class FeatureExampleAddCommandCommandValidator : AbstractValidator<FeatureExampleAddCommand>
{
    public FeatureExampleAddCommandCommandValidator()
    {
        RuleFor(x => x.FeatureExampleType)
            .IsInEnum();
    }
}
```

And then I have this Enum:

```
public enum FeatureExampleEnum
{
    EnumExample = 1,
}

```

But when I make a call with this json:

```
{
    "featureExampleType": 0
}
```
Or any other number that isn't on the enum, it doesn't return a bad request.
Works if I pass null.

Full example here: https://github.com/Arthur-Neto/template-website

**Comment by JeremySkinner** [comment]
Please provide a *small and minimal* example to reproduce the problem. Thanks.

**Comment by Arthur-Neto** [comment]
https://github.com/Arthur-Neto/fluentvalidation-issue1212

**Comment by JeremySkinner** [comment]
There are several problems with your example which is why it's not working.

First problem is you're not actually registering your validators. In your appication startup routing you should be calling `AddFluentValidation` after your call to `AddMvc`:

```csharp
services.AddMvc()
  .AddFluentValidation(fv => fv.RegisterValidatorsFromAssemblyContaining<ExampleModelValidator>())
  .SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
```

If you don't do this, then you won't get any automatic validation. [Please read the documentation.](https://fluentvalidation.net/aspnet)

Once you do this, the validator works as expected. Posting JSON of `{ "Type": 0 }` to the `ExampleController` returns the following 400 response:

```
{
    "errors": {
        "Type": [
            "'Type' has a range of values which does not include '0'."
        ]
    },
    "title": "One or more validation errors occurred.",
    "status": 400,
    "traceId": "0HLP4FKH9E47N:00000002"
}
```

As you can see, it correctly generated an error and returned a 400 response.

Your second issue is your test. Invoking a controller action method directly in a unit test *will not return a 400 response if validation fails*. The 400 response is not generated by the controller action (or by FluentValidation), but by ASP.NET Core's `ApiController` behaviours. During a regular HTTP request, this is generated after the controller executes. [Please read the documentation on the ApiController attribute here](https://docs.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-2.2#apicontroller-attribute) (specifically the section on automatic HTTP 400 responses). 

For unit tests, the only thing you can do is check `controller.ModelState.IsValid` is false, and that the modelstate dictionary contains the expected error message. If you want to see the full 400 response in a test, you need to use an integration test instead to simulate a full HTTP request/response. [You can read about that here](https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-2.2). 

To summarise:
- There is no issue with FluentValidation here.
- You need to register FluentValidation in your application startup routine.
- 400 responses generated by ASP.NET core's ApiController attribute after the controller is executed during an HTTP request. You can't simulate this in a unit test, only an integration test.

**Comment by Arthur-Neto** [comment]
Thanks for the answer, sorry for the inconvenience, just a mistake on my part, thought I had validated with other kinds of rules but I just tested here, should have search for more.
